snowflake/connect.go

```diff
@@ -0,0 +1,77 @@
+package snowflake
+
+// Originally from terraform-provider-snowflake/pkg/provider/provider.go
+import (
+	"github.com/chanzuckerberg/go-misc/keypair"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+	"github.com/snowflakedb/gosnowflake"
+)
+
+func ConfigureProvider(s *schema.ResourceData) (interface{}, error) {
```

Comment:
I'd avoid using the s *schema.ResourceData here and instead add your own struct -- that way this can be used outside the terraform context. Same with returning in interface{}, you could return the `db` type specifically instead